src/components/UserSelector.tsx

```diff
@@ -1,34 +1,25 @@
-import React, { useContext, useEffect, useState } from 'react';
+import { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
-import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { initUsers } from '../features/users';
+import { setAuthor } from '../features/author';
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
-
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+export const UserSelector = () => {
+  const dispatch = useAppDispatch();
+  const { users } = useAppSelector(state => state.users);
```

Comment:
You are accessing 'users' and 'author' individually. Consider using destructuring to access these properties. For example, you can write const { users, author } = useAppSelector(state => state).